src/core/thread_pool.hh

```diff
@@ -39,11 +83,11 @@ public:
     explicit thread_pool(reactor* r, sstring thread_name);
     ~thread_pool();
     template <typename T, typename Func>
-    future<T> submit(Func func) noexcept {
-        ++_aio_threaded_fallbacks;
+    future<T> submit(Func func, submit_reason reason = submit_reason::unknown) noexcept {
+        metrics.record_reason(reason);
         return inter_thread_wq.submit<T>(std::move(func));
     }
-    uint64_t operation_count() const { return _aio_threaded_fallbacks; }
```

Comment:
Yeah, summing all the reasons will have the same semantics as before. It's just not used anywhere else in the codebase so I removed it as dead code.

Reply:
I see, because the old metric is replaced entirely with the one the labels, makes sense. Not sure if seastar team will buy into not keeping the old metric (w/o labels) around but let's see.